src/support/string.cpp

```diff
@@ -132,7 +132,13 @@ std::ostream& printEscaped(std::ostream& os, std::string_view str) {
         if (c >= 32 && c < 127) {
           os << c;
         } else {
-          os << std::hex << '\\' << (c / 16) << (c % 16) << std::dec;
+          if (mode == EscapeMode::Normal) {
+            os << std::hex << '\\' << (c / 16) << (c % 16) << std::dec;
+          } else if (mode == EscapeMode::JSON) {
+            os << std::hex << "\\u00" << (c / 16) << (c % 16) << std::dec;
```

Comment:
JSON requires exactly 4 hex digits representing a UTF-16 code unit (i.e. 2 bytes at once), so it's going to be slightly more complicated than this, unfortunately.